doc/bn.tex

```diff
@@ -820,28 +820,32 @@ \subsection{Single Digit}
 \}
 \end{alltt} \end{small}
 
-\subsection{Long Constants}
+\subsection{Int32 and Int64 Constants}
 
-To set a constant that is the size of an ISO C ``unsigned long'' and larger than a single digit the following function
-can be used.
+To set a constant with 32 or 64 bits the following function can be used.
 
-\index{mp\_set\_int}
+\index{mp\_set\_sint}
 \begin{alltt}
-int mp_set_int (mp_int * a, unsigned long b);
+void mp_set_sint (mp_int * a, int32_t b);
+void mp_set_uint (mp_int * a, uint32_t b);
+void mp_set_sint64 (mp_int * a, int64_t b);
+void mp_set_uint64 (mp_int * a, uint64_t b);
 \end{alltt}
 
-This will assign the value of the 32-bit variable $b$ to the mp\_int $a$.  Unlike mp\_set() this function will always
-accept a 32-bit input regardless of the size of a single digit.  However, since the value may span several digits
-this function can fail if it runs out of heap memory.
-
-To get the ``unsigned long'' copy of an mp\_int the following function can be used.
+This will assign the value of the variable $b$ to the mp\_int $a$. The unsigned variants set a positive value, the signed variants set the sign. The value can be obtained again by calling the following function.
 
 \index{mp\_get\_int}
 \begin{alltt}
-unsigned long mp_get_int (mp_int * a);
+int32_t mp_get_sint (mp_int * a);
+uint32_t mp_get_uint (mp_int * a);
+uint32_t mp_get_mag (mp_int * a);
+int64_t mp_get_sint64 (mp_int * a);
+uint64_t mp_get_uint64 (mp_int * a);
+uint64_t mp_get_mag64 (mp_int * a);
 \end{alltt}
 
-This will return the 32 least significant bits of the mp\_int $a$.
+This will return either 32 or 64 least significant bits of the mp\_int $a$. The unsigned functions
```

Comment:
There is more than one function listed and the return only one specific type, not several.

"These functions return the 32 and 64 least significant bits of \texttt{mp\_int a} respectively."

I used "\texttt{}" because it is code but it is quite a mess in the whole documentation, we should standardize it a bit.
But that is work and on top of it: it is documentation! I know ;-)

Reply:
fixed